gno.land/pkg/sdk/vm/keeper.go

```diff
@@ -205,6 +205,7 @@ var gnoStoreContextKey gnoStoreContextKeyType
 func (vm *VMKeeper) newGnoTransactionStore(ctx sdk.Context) gno.TransactionStore {
 	base := ctx.Store(vm.baseKey)
 	iavl := ctx.Store(vm.iavlKey)
+	vm.gnoStore.SetGasMeter(ctx.GasMeter())
```

Comment:
This is unsafe, because vm.gnoStore is shared, and this would set the gas meter globally; ie. not scoped to the transaction.

Reply:
This is safe under the current ABCI protocol implementation. All ABCI requests to the app are synchronized at the ABCI protocol level. In other words, the VM Keeper processes one request at a time, which is why gnoStore can be shared.

Reply:
This is safe under the current ABCI protocol implementation. All ABCI requests to the app are synchronized at the ABCI protocol level. In other words, the VM Keeper processes one request at a time, which is why gnoStore can be shared.

The GasMeter is created and wrapped in each transaction context before being set in the gnoStore.

Let me know if you still think this not safe.

Reply:
@thehowl  good catch! I will pass the gasMeter to gnoStore.BeginTransaction() instead